lib/authentication/auth_okta.js

```diff
@@ -49,47 +49,38 @@ function AuthOkta(password, region, account, clientType, clientVersion, httpClie
   * @returns {null}
   */
   this.authenticate = async function (authenticator, serviceName, account, username) {
-    let ssoUrl;
-    let tokenUrl;
-    await step1(authenticator, serviceName, account, username).then((response) => {
-      const responseData = response['data'];
-      const success = responseData['success'];
-      const errorCode = responseData['code'];
-      const errorMessage = responseData['message'];
-
-      if (typeof success === 'undefined' || errorCode === 'undefined' || errorMessage === 'undefined') {
-        throw new Error('Unable to use provided Okta address as an authenticator. Is the authenticator URL correct?');
-      }
+    const response = await step1(authenticator, serviceName, account, username);
+    const responseData = response['data'];
+    const success = responseData['success'];
+    const errorCode = responseData['code'];
+    const errorMessage = responseData['message'];
+
+    if (typeof success === 'undefined' || errorCode === 'undefined' || errorMessage === 'undefined') {
+      throw new Error('Unable to use provided Okta address as an authenticator. Is the authenticator URL correct?');
+    }
 
-      if (success !== true) {
-        throw new Error(`Unable to use provided Okta address as an authenticator. Error code: ${errorCode}, error message: ${errorMessage}`);
-      }
+    if (success !== true) {
+      throw new Error(`Unable to use provided Okta address as an authenticator. Error code: ${errorCode}, error message: ${errorMessage}`);
+    }
 
-      ssoUrl = responseData['data']['ssoUrl'];
-      tokenUrl = responseData['data']['tokenUrl'];
-    });
+    const ssoUrl = responseData['data']['ssoUrl'];
+    const tokenUrl = responseData['data']['tokenUrl'];
 
     step2(authenticator, ssoUrl, tokenUrl);
-
-    let oneTimeToken;
-    await step3(tokenUrl, username, password).then((response) => {
-      const data = response['data'];
-
-      if (data['sessionToken']) {
-        oneTimeToken = data['sessionToken'];
-      } else {
-        oneTimeToken = data['cookieToken'];
-      }
-    });
-
-    let responseHtml;
-    await step4(oneTimeToken, ssoUrl).then((response) => {
-      responseHtml = response['data'];
-    });
+    
+    const responseHtml = await step4( await step3(tokenUrl, username, this.password), ssoUrl);
 
     step5(responseHtml);
   };
 
+  this.reauthenticate = async function (body) {
+    await this.authenticate(connectionConfig.getAuthenticator(),
```

Comment:
Is it possible to use the current okta authenticator instead of creating a new one that uses the same credentials?

Reply:
I checked that we do not need to run step 1 and 2 when the driver reauthenticates. As a result, I fixed the driver to run only step 3 to 5 for renewing the token. Also, I realized that the auth class does not need to have credential info as public, so I refactored some codes. Please have a look.

Reply:
Looks good